src/aus.h

```diff
@@ -0,0 +1,329 @@
+    class aus_data
+    {
+
+    public:
+        aus_data()
+        {
+            _start_time = std::chrono::system_clock::to_time_t(std::chrono::system_clock::now());
+            _librealsense_version = RS2_API_VERSION_STR;
+            _os_name = get_os_name();
+            _platform_name = PLATFORM;
+
+        }
+
+        std::string get_os_name()
+        {
+            #ifdef _WIN32
+            return "Windows";
+            #else
+            #ifdef __APPLE__
+            return "Mac OS";
+            #else
+            #ifdef __linux__
+            return "Linux";
+            #else
+            return "Unknown";
+            #endif
+            #endif
+            #endif
+        }
+
+        
+        void set(std::string key, long value)
+        {
+            if (_mp.find(key) != _mp.end())
+            {
+                _mp[key]->set(value);
+            }
+            else
+            {//_mp[key] = std::make_unique<aus_counter>(new aus_counter(value));
+                _mp[key] = new aus_counter(value);
+            }
+
+        }
+
+        void increment(std::string key)
+        {
+            if (_mp.find(key) != _mp.end())
+            {
+                _mp[key]->increment();
+            }
+            else
+            {
+               // _mp[key] = std::make_unique<aus_counter>(new aus_counter(1));
+                _mp[key] = new aus_counter(1);
+            }
+        }
+
+        void decrement(std::string key)
+        {
+            if (_mp.find( key ) != _mp.end() && _mp[key] >0)
+            {
+                _mp[key]->decrement();
+            }
+        }
+
+        long get(std::string key)
+        {
+            assert_key_exists(key);
+            return _mp[key]->get();
+        }
+
+        void start(std::string key)
+        {
+            if (_mp.find(key) == _mp.end())
+            {
+                _mp.insert(std::make_pair(key, new aus_timer()));
+                _mp[key]->start();
+            }
+            else
+            {
+                _mp[key]->start();
+            }
+        }
+
+        void stop(std::string key)
+        {
+            assert_key_exists(key);
+            _mp[key]->stop();
+        }
+
+        std::vector<std::string> get_counters_list()
+        {
+            std::vector<std::string> result;
+            for (const auto& pair : _mp) {
+                result.push_back(pair.first);
+            }
+            return result;
+        }
+
+        bool device_exist( std::string serial )
+        {
+            if ( _mp_devices_manager.find( serial ) == _mp_devices_manager.end())
+            {
+                return false;
+            }
+            return true;
+        }
+
+        void insert_device_to_device_manager( std::string serial, std::string name )
+        {
+            increment( librealsense::aus_build_system_counter_name( "CONNECTED_DEVICES" ));
+            increment( librealsense::aus_build_system_counter_name( "CONNECTED_DEVICES", name ) );
```

Comment:
Why do we increment both `"CONNECTED_DEVICES"` and `CONNECTED_DEVICES", name`

Reply:
?

Reply:
we increment the amount of total devices and the amount of devices of a specific camera